components/tidb_query_expr/src/impl_time.rs

```diff
@@ -700,6 +700,38 @@ pub fn add_duration_and_string(
     Ok(Some(res))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn duration_duration_time_diff(
+    ctx: &mut EvalContext,
+    arg1: &Duration,
+    arg2: &Duration,
+) -> Result<Option<Duration>> {
+    let res = match arg1.checked_sub(*arg2) {
+        Some(res) => res,
+        // `check_sub` returns `None` if the sub operation overflow/underflow i64 bound or mysql_time_value bound.
+        // and we need to treat these two case separately.
+        // if `arg1 - arg2` is in [`MAX_NANOS`, `i64::MAX`], return max value of mysql `TIME` type.
```

Comment:
Should be (`MAX_NANOS`, `i64::MAX`) here?

Reply:
You are right. The `if` statement and comment is not exactly correct. When reaching the `None` case, sub result can be neither `MAX_NANOS ` nor `-MAX_NANOS`. And furthermore, `i64::MAX` and `i64::MIN` should be considered as valid. So the range should be (`MAX_NANOS`, `i64::MAX`] and [`i64::MAX`,`-MAX_NANOS`).